### DISCUSSION TITLE
reconnect automatically to 4G

### DISCUSSION THREAD

**Discussion by iznobe** [Salamek/huawei-lte-api#131] [Ideas]
Hi , i would like to know if it is possible to have a script pinging every number of minutes and if no internet detected , reconnect automatically the routeur to 4G ?

i have a B818.263 and ping and reconnect , are included in WEB GUI , so i think it is possible to do this .
but , i dont know how ...

**Comment by Salamek** [comment]
Hi @iznobe yes it should be possible, 
i would suggest to do the "pinging" in python script and just call `reconnect` endpoint when `ping` fails multiple times. Actually this usage was initial reason why this library was created :-D

I have whole private/proprietary project doing this exact thing, it you are interested i can release it under GPL for public use?

**Reply by iznobe** [reply]
Hi Salamek , it would be wonderful :D  :+1:

**Comment by Salamek** [comment]
Ok i have released the code under GPL-3 here https://github.com/Salamek/netkeeper it is renamed/cleared it builds...
But i only tested netkeeper status command...
